lib/espejote.libsonnet

```
@@ -65,8 +66,195 @@ local managedResource(name, namespace) = {
   },
 };
 
+
+local generateRolesForManagedResource(manifest) =
+  local manifestMeta = std.get(manifest, 'metadata', {});
+  assert std.get(manifestMeta, 'name') != null : 'name is required';
+  assert std.get(manifestMeta, 'namespace') != null : 'namespace is required';
+  local manifestSpec = std.get(manifest, 'spec', {});
+
+  local clusterScoped(resource) =
+    if std.get(resource, '_namespaced') != null then
+      !std.get(resource, '_namespaced')
+    else
```

this could be the following afaict:

```suggestion
    if std.objectHas(resource, '_namespaced') then
      !resource._namespaced
    else
```